src/events/guildCreate.ts

```diff
@@ -29,6 +29,10 @@ export default (async function run(guild) {
     .setThumbnail(client.user.displayAvatarURL())
     .setColor(genColor(200));
 
-  await new Commands(client).registerCommandsForGuild(guild);
-  if (dmChannel) await dmChannel.send({ embeds: [embed] });
+  await guild.commands.set(commands.map(command => command.data));
+  try {
+    if (dmChannel) await dmChannel.send({ embeds: [embed] });
```

Comment:
Maybe also send a message in the welcoming channel for quick setup ? Dms aren't working sometimes

Reply:
> how does the bot locate the welcome channel

I believe you would use `interaction.guild.systemChannel` to get the system/welcome channel.

Reply:
DONE